apps/daimo-mobile/src/view/screen/send/SendNoteScreen.tsx

```diff
@@ -51,19 +56,65 @@ export function SendNoteScreen({ route }: Props) {
   }, [nav, amountChosen]);
   useDisableTabSwipe(nav);
 
+  const [account] = useAccount();
+  assert(account != null);
+
+  const [externalAction, setExternalAction] = useState<ExternalAction>({
+    type: "share",
+    exec: shareSheetURL,
+  });
+
+  useEffect(() => {
+    if (!recipient) return;
+
+    const testAndGetComposer = async (): Promise<ExternalAction> => {
```

Comment:
i think you can pull this into its own function below the componet

```
async function getSendLinkAction(recipient: Contact): Promise<ExternalAction> {
 // ...
}
```

less logic inside component always nice

Reply:
done